### ISSUE TITLE
Plug-in applies incorrectly to .d.ts/.d.mts files

### ISSUE THREAD

**Issue by sentience** [egoist/rollup-plugin-hashbang#36]
When this plugin is added to Rollup and Rollup is configured to emit .d.ts/.d.mts type declaration files, the build may fail due to an apparent race between the creation of the declaration files on disk and the invocation of chmod to set their executable flag to +x. Unless I'm mistaken, we don't need to make these type declaration files executable anyway, so the best fix would probably just be to exclude these files.

Additionally, this plug-in is preserving the shebang line in .d.ts/.d.mts files, causing it to produce type declaration files like this:
```js
#!/usr/bin/env node
```

Without this plug-in, those same type declaration files contain this:
```js

export { }

```

In short, rollup-plugin-hashbang should ignore **.d.(m)ts** files generated by Rollup.

See also downstream issue egoist/tsup#1001
